tests/functional/vault/test_config.py

```diff
@@ -43,32 +45,174 @@ def test_vault_name_symbol_override(guardian, gov, rewards, token, Vault):
 
 
 @pytest.mark.parametrize(
-    "getter,setter,val",
+    "getter,setter,val,gov_allowed,mgmt_allowed,guard_allowed",
     [
-        ("name", "setName", "NewName yVault"),
-        ("symbol", "setSymbol", "yvNEW"),
-        ("emergencyShutdown", "setEmergencyShutdown", True),
-        ("guardian", "setGuardian", None),
-        ("rewards", "setRewards", None),
-        ("management", "setManagement", None),
-        ("performanceFee", "setPerformanceFee", 1000),
-        ("managementFee", "setManagementFee", 1000),
-        ("depositLimit", "setDepositLimit", 1000),
+        ("name", "setName", "NewName yVault", True, False, False),
+        ("symbol", "setSymbol", "yvNEW", True, False, False),
+        ("emergencyShutdown", "setEmergencyShutdown", True, True, False, True),
+        ("guardian", "setGuardian", None, True, False, True),
+        ("rewards", "setRewards", None, True, False, False),
+        ("management", "setManagement", None, True, False, False),
+        ("performanceFee", "setPerformanceFee", 1000, True, False, False),
+        ("managementFee", "setManagementFee", 1000, True, False, False),
+        ("depositLimit", "setDepositLimit", 1000, True, False, False),
     ],
 )
-def test_vault_setParams(gov, vault, rando, getter, setter, val):
+def test_vault_setParams(
+    chain,
+    gov,
+    guardian,
+    management,
+    vault,
+    rando,
+    getter,
+    setter,
+    val,
+    gov_allowed,
+    mgmt_allowed,
+    guard_allowed,
+):
     if not val:
         # Can't access fixtures, so use None to mean any random address
         val = rando
 
-    # Only governance can set this param
+    # rando shouldn't be able to call these methods
     with brownie.reverts():
         getattr(vault, setter)(val, {"from": rando})
 
+    if guard_allowed:
+        getattr(vault, setter)(val, {"from": guardian})
+        assert getattr(vault, getter)() == val
+        chain.undo()
+    else:
+        with brownie.reverts():
+            getattr(vault, setter)(val, {"from": guardian})
+
+    if mgmt_allowed:
+        getattr(vault, setter)(val, {"from": management})
+        assert getattr(vault, getter)() == val
+        chain.undo()
+    else:
+        with brownie.reverts():
+            getattr(vault, setter)(val, {"from": management})
+
     getattr(vault, setter)(val, {"from": gov})
     assert getattr(vault, getter)() == val
 
 
+@pytest.mark.parametrize(
+    "key,setter,val,gov_allowed,mgmt_allowed,guard_allowed",
+    [
+        ("debtLimit", "updateStrategyDebtLimit", 500, True, True, False),
+        ("rateLimit", "updateStrategyRateLimit", 10, True, True, False),
+    ],
+)
+def test_vault_updateStrategy(
```

Comment:
Guardian is never allowed on these functions, and gov/mgmt can always both call them. Perhaps not necessary here.